compendium/DeclarativeServices/test/TestUpdateConfiguration.cpp

```diff
@@ -113,9 +103,107 @@ TEST_F(tServiceComponent, testUpdateConfig_Modified) //DS_CAI_FTC_1
+            scr::dto::ComponentState::UNSATISFIED_REFERENCE)
+    << "Component state should be UNSATISIFIED_REFERENCE";
+
+  // Get a service reference to ConfigAdmin to create the component instance.
+  auto configAdminService =
+    GetInstance<cppmicroservices::service::cm::ConfigurationAdmin>();
+  ASSERT_TRUE(configAdminService) << "GetService failed for ConfigurationAdmin";
+
+  auto configObject = configAdminService->GetConfiguration(componentName);
+  auto configObjInstance = configObject->GetPid();
+  cppmicroservices::AnyMap props(
+    cppmicroservices::AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
+  configObject->Update(props);
+
+  // wait for the asynchronous task to take effect
+  auto result = RepeatTaskUntilOrTimeout(
+    [&compDescDTO, &compConfigs, this, &testBundle, &componentName]() {
+      compDescDTO =
+        dsRuntimeService->GetComponentDescriptionDTO(testBundle, componentName);
+      if (compDescDTO.name != "") {
+        compConfigs =
+          this->dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+      }
+    },
+    [&compConfigs]() -> bool {
+      return compConfigs.at(0).state == scr::dto::ComponentState::SATISFIED;
+    });
+
+  ASSERT_TRUE(result)
+    << "Timed out waiting for component state to become SATISFIED.";
+
+  //Request a service reference to the new component instance.
+  auto service = GetInstance<test::CAInterface>();
+  ASSERT_TRUE(service) << "GetService failed for CAInterface";
+
+  compConfigs = GetComponentConfigs(testBundle, componentName, compDescDTO);
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
+    << "Factory instance state should be ACTIVE";
+
+  //Update property
+  const std::string instanceId{ "instance1" };
+  props["uniqueProp"] = instanceId;
+  configObject->Update(props);
+
+  result = RepeatTaskUntilOrTimeout(
+    [&compDescDTO, &compConfigs, this, &testBundle, &configObjInstance]() {
+      compDescDTO = dsRuntimeService->GetComponentDescriptionDTO(
+        testBundle, configObjInstance);
+      if (compDescDTO.name != "") {
+        compConfigs =
+          this->dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+      }
+    },
+    [&compConfigs, &instanceId]() -> bool {
+      if (compConfigs.size() == 1) {
+        auto properties = compConfigs.at(0).properties;
+        auto id = properties.find("uniqueProp");
+        if (id != properties.end()) {
+          return (id->second == instanceId);
+        }
+      }
+      return false;
+    });
+
+  ASSERT_TRUE(result) << "Timed out waiting for Update Configuration"
+                         "to complete.";
+
+  compConfigs = GetComponentConfigs(testBundle, componentName, compDescDTO);
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::SATISFIED)
+    << "Component state should be SATISFIED";
```

Comment:
DS is supposed to log the exception thrown by the Modified method. Is there a way to confirm that in a test?

Reply:
you could implement a LogService for the test and use that to verify that an exception was thrown.